### PR TITLE
Add CI to contributing

### LINKED ISSUE #695: Add precommit ci to CONTRIBUTING.md
I wonder if we also want to say something about `precommit-ci`, as either a separate or nested topic for `pre-commit hooks` in general, in the contributing file?

_Originally posted by @sjspielman in https://github.com/AlexsLemonade/scpca-nf/issues/694#issuecomment-1942088757_
            
---

Something to circle back to!

### PR DESCRIPTION
**Pull request by sjspielman**
Closes #695 

Here I've added a small section about CI into contributing. I nested this inside the PRs header and mentioned both required workflows and the pre-commit ci workflow (do I still have it correct that passing is not mandatory? I don't have access to Settings here so couldn't tell for sure).

Let me know if this is the right place in `CONTRIBUTING.md` to have added this, as well as any details to be expanded upon or reduced.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -20,6 +20,19 @@ To allow for efficient review, please include in any pull request a concise and
...
+There are several automatic checks performed by GitHub Actions in all pull requests filed to `main` or `development`.
+
+These checks are required to pass before pull requests can be merged:
+
+- [Spell check R Markdown and Markdown files](.github/workflows/spell-check.yml): This workflow ensures there are no spelling errors in R Markdown and Markdown files.
+- [Check Nextflow stub](.github/workflows/nextflow-stub-check.yaml): This workflow ensures that the [stub workflow](#stub-workflows) runs without errors.
+- [Check Nextflow config](.github/workflows/nextflow-config-check.yaml): This workflow ensures that there are no syntax errors in the Nextflow configuration files
+
+There is also one additional `pre-commit ci` workflow which runs all [pre-commit hooks as described in this section](#pre-commit-hooks), except for the spell check pre-commit hook.
+Although highly recommended, it is not required that this workflow passes before pull requests can be merged.
```
Comment (jashapiro) [inline CONTRIBUTING.md:34]: Spell check is not required to pass either, the way this repo is currently set up.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -20,6 +20,19 @@ To allow for efficient review, please include in any pull request a concise and
...
 
+### Continuous integration in pull requests
+
+There are several automatic checks performed by GitHub Actions in all pull requests filed to `main` or `development`.
+
+These checks are required to pass before pull requests can be merged:
+
+- [Spell check R Markdown and Markdown files](.github/workflows/spell-check.yml): This workflow ensures there are no spelling errors in R Markdown and Markdown files.
+- [Check Nextflow stub](.github/workflows/nextflow-stub-check.yaml): This workflow ensures that the [stub workflow](#stub-workflows) runs without errors.
+- [Check Nextflow config](.github/workflows/nextflow-config-check.yaml): This workflow ensures that there are no syntax errors in the Nextflow configuration files
```
Comment (jashapiro) [inline CONTRIBUTING.md:31]: Switching order here for alphabetical/consistency with display reasons.
```suggestion
- [Check Nextflow config](.github/workflows/nextflow-config-check.yaml): This workflow ensures that there are no syntax errors in the Nextflow configuration files
- [Check Nextflow stub](.github/workflows/nextflow-stub-check.yaml): This workflow ensures that the [stub workflow](#stub-workflows) runs without errors.
- [Spell check R Markdown and Markdown files](.github/workflows/spell-check.yml): This workflow ensures there are no spelling errors in R Markdown and Markdown files.
```

**Review by jashapiro — APPROVED** [review]
Placement here looks fine. I'd just maybe rephrase about what is required, as spelling is not.

File: CONTRIBUTING.md (hunk 1 above)
Comment (sjspielman) [reply CONTRIBUTING.md:34]: Ah, didn't know that! Also looks like the precommit ci only runs on `development` (if i'm reading the yaml syntax correctly), so will change that too.

Comment (jashapiro) [reply CONTRIBUTING.md:32]: No, it runs everywhere. That is where it files its PRs to (when updating hook versions)
